src/commands/ingress/toggleIngressVisibility/toggleIngressVisibility.ts

```diff
@@ -0,0 +1,39 @@
+import type { Ingress } from "@azure/arm-appcontainers";
+import { AzureWizard, AzureWizardExecuteStep, IActionContext, createSubscriptionContext, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { IngressConstants } from "../../../constants";
+import type { ContainerAppItem } from "../../../tree/ContainerAppItem";
+import type { IngressItem } from "../../../tree/IngressItem";
+import { createActivityContext } from "../../../utils/activityUtils";
+import { localize } from "../../../utils/localize";
+import { pickContainerApp } from "../../../utils/pickContainerApp";
+import type { IngressContext } from "../IngressContext";
+import { ToggleIngressVisibilityStep } from "./ToggleIngressVisibilityStep";
+
+export async function toggleIngressVisibility(context: IActionContext, node?: IngressItem | ContainerAppItem): Promise<void> {
+    const { subscription, containerApp } = node ?? await pickContainerApp(context);
+
+    const wizardContext: IngressContext = {
+        ...context,
+        ...createSubscriptionContext(subscription),
+        ...(await createActivityContext()),
+        subscription,
+        containerApp
+    };
+
+    const ingress: Ingress = nonNullValueAndProp(containerApp.configuration, 'ingress');
+    const title: string = localize('toggleIngressVisibility', 'Toggle ingress visibility to "{0}" for container app "{1}"', ingress.external ? IngressConstants.internal : IngressConstants.external, containerApp.name);
+
+    const executeSteps: AzureWizardExecuteStep<IngressContext>[] = [
+        new ToggleIngressVisibilityStep()
+    ];
+
+    const wizard: AzureWizard<IngressContext> = new AzureWizard(wizardContext, {
+        title,
+        executeSteps,
+        showLoadingPrompt: true
+    });
+
+    // Title normally gets set during prompt phase... since no prompt steps are provided we must set the 'activityTitle' manually
+    wizardContext.activityTitle = title;
+    await wizard.execute();
```

Comment:
Just as a sidenote, I prefer having the `promptStep` actually be the wizard step, and then just having the "execute" code just be part of the command.

Reply:
See section above, but I created a base execute step that allowed all ingress steps to inherit a method for updating ingress settings.  I think that this case was a bit of an outlier in that it still had a smidge of a prompt inside the execution, but normally it doesn't... and in this case I can just move the warning prompt out to keep it a bit more consistent.